Tests/Amr/Advection_AmrLevel/Source/AmrLevelAdv.cpp

```diff
@@ -710,6 +710,11 @@ AmrLevelAdv::errorEst (TagBoxArray& tags,
 {
     MultiFab& S_new = get_new_data(Phi_Type);
 
+    // Create temporary multifab with properly filled patches
+    const Real cur_time = state[Phi_Type].curTime();
+    MultiFab phi(S_new.boxArray(), S_new.DistributionMap(), NUM_STATE, 1);
+    FillPatch(*this, phi, phi.nGrow(), cur_time, Phi_Type, 0, NUM_STATE);
```

Comment:
We do not need to create a new temporary MulitFab here.  It's OK to call FillPatch on S_new.  We also do not need to fill all ghost cells, just one is sufficient for computing grad phi.  Finally, if level >= max_phigrad_lev, we do not need to call FillPatch.

Reply:
I can't seem to find a way around not creating a temporary `MultiFab` for the gradient check (assertion error trying to fill in ghost cells). I think I can only implement the latter two.

Reply:
OK.  Didn't realize the StateData in this example does not have ghost cells.